engine/src/game_type.rs

```diff
@@ -19,30 +21,33 @@ impl Default for GameType {
 }
 
 impl GameType {
-    pub fn from_str(str: &str) -> GameType {
+    pub fn from_str(str: &str) -> Result<GameType, GameError> {
```

Comment:
if we move this (and the other parsing-type functions for bugs/etc) into an [`impl FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html), we can use the std library's `parse` for these, which is a little cleaner

Reply:
Good idea, but I will address this is a future PR see #6